### PR TITLE
PaymentTypeController PaymentType Model

### LINKED ISSUE #1: Allow developers to access the Payment Type resource
Verbs to be supported

1. GET
1. POST
1. PUT
1. DELETE

User should be able to GET a list, and GET a single item.

### PR DESCRIPTION
**Pull request by gretchenmarie**
## Link to Ticket
Closes #1 

## Description of Proposed Changes
- Created model for PaymentTypes
- Created mvc for PaymentTypes

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout <gw-paymenttype>
Add filler paymentType data into your SQL
Test POST, PUT, DELETE, and GET in POSTMAN
```
Run SQL to seed your database and run in Poastman
There is a Foreign Key constraint in Orders-you will need to take order data out of Sql in order to run the Delete method without errors.

## Impacted Areas in Application

List general components of the application that this PR will affect:

* PaymentTypeModel and PaymentTypeModelController

## Mentions @username

@Akeen11 
@KaylaReid 
@askingalot 

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.

### CODE REVIEW

**Review by KaylaReid — COMMENTED** [review]
Looks great

**Review by Akeen11 — APPROVED** [review]
👍

**Review by KaylaReid — APPROVED** [review]
looks good

File: Bangazon/Controllers/PaymentTypeController.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            }
+
+        }
+
+        // PUT api/PaymentType/Put
+        //Put method changes a current object
+        [HttpPut("{id}")]
+        public async Task<IActionResult> Put([FromRoute] int id, [FromBody] PaymentType paymentType)
+        {
+            string sql = $@"
```
Comment (askingalot) [inline Bangazon/Controllers/PaymentTypeController.cs:94]: Add `AcctNumber` and `CustomerId` to the update.

File: Bangazon/Controllers/PaymentTypeController.cs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        // POST api/PaymentType/Post
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] PaymentType paymentType)
+        {
+
+            string sql = $@"INSERT INTO PaymentType
+            (AcctNumber, Name, CustomerId)
+            VALUES
+            ('{paymentType.AcctNumber}', '{paymentType.Name}','{paymentType.CustomerId}');
+            select MAX(Id) from PaymentType";
```
Comment (askingalot) [inline Bangazon/Controllers/PaymentTypeController.cs]: Change the select to `SELECT SCOPE_IDENTITY()`

**Review by askingalot — CHANGES_REQUESTED** [review]

**Review by askingalot — APPROVED** [review]

**Pull request merged**